lldb/include/lldb/Core/Telemetry.h

```diff
@@ -28,6 +29,17 @@
 namespace lldb_private {
 namespace telemetry {
 
+struct LLDBConfig : public ::llvm::telemetry::Config {
+  // If true, we will collect full details about a debug command (eg., args and
+  // original command). Note: This may contain PII, hence can only be enabled by
+  // the vendor while creating the Manager.
+  const bool m_detailed_command_telemetry;
```

Comment:
I like the idea of the Configuration being the one deciding what events we emit. That's easy to audit. Can we make this a bit more generic and maybe have a set of `telemetry::KindType` that are enabled? We can have a default configuration that enables everything by default upstream (for testing) and then vendors have to override this downstream with the events they're interested in. Originally I imagined doing this in the emitter, but I think the config is a better place to do that. WDYT?

Reply:
This is for controlling how much data we collect for a given command - not for which type of events we collect.

But we can also add that if you want. Although you could already do taht by overriding the `::dispatch()` to simply return/ignore the kind of entries you dont want to record. At that point, the data had been collected but it's not "leaving" LLDB yet - so still fne? (Or do you want something stricter? eg., not collect the data at all?)

Reply:
> Ack, I think this is fine as is. Given that this is a struct, please drop the m_ prefix.
done